backend/aws_helpers/dynamo_db_utils/execution_db.py

```diff
@@ -1,17 +1,22 @@
+import json
+
 from dataclasses import dataclass
+from datetime import datetime
+
 from backend.aws_helpers.dynamo_db_utils.base_db import BaseData, BaseDDBUtil, enumclass, changevar
 from backend.common.constants import EXECUTION_TABLE_NAME, AWS_REGION
+from typing import Union
 
 @dataclass
 class ExecutionData(BaseData):
     """Data class to hold the attribute values of a record of the execution-table DynamoDB table"""
     execution_id: str
-    user_id: str
-    name: str
-    timestamp: str
-    data_source: str
-    status: str
-    progress: int
+    user_id: str = None
+    name: str = None
+    timestamp: str = None
+    data_source: str = None
+    status: str = None
+    progress: int = None
```

Comment:
I have to leave all but execution_id as nullable to allow getting an entry using only an execution_id

Reply:
understood. that's ok